sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -47,17 +51,52 @@ def lambda_handler(event, context):
         filename = record["s3"]["object"]["key"]
 
         logger.info(f"Attempting to insert {os.path.basename(filename)} into database")
-        # TODO: change below logics to use new FilenameParser
-        # when we create schema and write file metadata to DB
         filename_parsed = FilenameParser(filename)
-        filename_parsed.upload_filepath()
-        metadata = None
+        filepath = filename_parsed.upload_filepath()
 
-        # TODO: remove this check since upload api validates filename?
-        # Found nothing. This should probably send out an error notification
-        # to the team, because how did it make its way onto the SDS?
-        if metadata is None:
-            logger.info("Found no matching file types to index this file against.")
-            return None
+        # confirm that the file is valid
+        if filepath["statusCode"] == 400:
+            logger.error(filepath["body"])
+            break
```

Comment:
```suggestion
        if filepath["statusCode"] != 200:
            logger.error(filepath["body"])
            break
```
Do you want to raise here instead of breaking? Or return the bad statusCode and message?